server/src/main/scala/org/http4s/armeria/server/ArmeriaHttp4sHandler.scala

```diff
@@ -181,17 +191,19 @@ private[armeria] class ArmeriaHttp4sHandler[F[_]](
         .toList
     )
 
-  /** Converts an HTTP payload to [[EntityBody]]. */
-  private def toBody(req: HttpRequest): EntityBody[F] =
-    if (canHasBody(req.method()))
-      req
-        .toStreamBuffered[F](1)
-        .flatMap { obj =>
-          val data = obj.asInstanceOf[HttpData]
-          Stream.chunk(Chunk.array(data.array()))
-        }
-    else
-      EmptyBody
+  /** Converts an HTTP payload to [[Entity]]. */
+  private def toEntity(req: HttpRequest): Entity[F] =
+    if (canHasBody(req.method())) {
+      Entity.Default(
+        req
+          .toStreamBuffered[F](1)
+          .flatMap { obj =>
+            val data = obj.asInstanceOf[HttpData]
+            Stream.chunk(Chunk.array(data.array()))
+          },
+        None)
```

Comment:
Should we the length of the content to the `Entity` if exists?
For example:
```scala
val contentLength = req.headers().contentLength()
val length = 
  if (contentLegnth <= 0) {
      None
  } else {
     Some(contentLength)
  }
```

Reply:
I can say that for now `http4s` almost don't use the `Entity.Default#length` on the server side. `None` here is considered likely as 'the entity length isn't defined'. Because, I guess, trusting `Content-Length` isn't reliable :)